### PR TITLE
Modified EnumPrisma.stencil for extending only necessary enums

### PR DESCRIPTION
**Pull request by AndriiDoroshko**
Implemented  `public protocol GeneratePrism {}` for enums which needs to be extended by EnumPrism. In EnumPrism.stencil I've deleted setters and changed `for enum in types.enums where enum.cases.all.count > 0 and not enum.accessLevel == 'private'` to `for enum in types.implementing.GeneratePrism|enum`. 
Now, all enums marked `GeneratePrism` are extending by EnumPrism

### CODE REVIEW

File: PlayerControls/sources/AdVideoControls.swift [hunk 1]
```diff
@@ -75,7 +75,7 @@ public final class AdVideoControls: UIViewController {
         public let isLoading: Bool
         public let airplayActiveViewHidden: Bool
         
-        public enum MainAction {
+        public enum MainAction: GeneratePrism {
```
Comment (AndriiMoskvin) [inline PlayerControls/sources/AdVideoControls.swift]: Maybe just `Prism`?

@AlexeyDemedetskiy Your thoughts?

Comment (AndriiDoroshko) [reply PlayerControls/sources/AdVideoControls.swift]: In my opinion, `GeneratePrism` means "generate prism for this Enum", but `Prism` doesn't sound so evident

Comment (AndriiMoskvin) [reply PlayerControls/sources/AdVideoControls.swift]: Nobody names protocols like this. `Prism` is pretty self explanatory. And this is empty protocol - so the only purpose of it - marking.

File: PlayerControls/sources/ContentControlsViewController.swift [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 import Foundation
 
+public protocol GeneratePrism {}
```
Comment (BogdanBilonog) [inline PlayerControls/sources/ContentControlsViewController.swift]: Add comment describing this protocol

File: PlayerControls/templates/EnumPrism.stencil [hunk 3]
```diff
@@ -46,34 +46,21 @@ import Foundation
         @available(*, unavailable, message:"Optional associated value is not supported yet")
     {%endif%}
     public var {{case.name}}: {% if case.associatedValues.all.count > 1 %}{% call associatedValueVarMultiType case %}{% else %}{% call associatedValueVarUnitType case %}{% endif %} {
-        get {
             guard case let .{{ case.name }}({{ case.name }}) = self else { return nil }
```
Comment (BogdanBilonog) [inline PlayerControls/templates/EnumPrism.stencil]: Fix indent

File: PlayerControls/templates/EnumPrism.stencil [hunk 4]
```diff
@@ -46,34 +46,21 @@ import Foundation
...
-            self = .{{ case.name }}({% if case.associatedValues.all.count > 1 %}{% call associatedValueVarMultiSet case newValue %}{% else %}{% call associatedValueVarUnitSet case newValue %}{% endif %})
-        }
     }
 {% endmacro %}
 
 /*Creates Bool var with given simple case*/
 {% macro simpleVar case %}
     public var is{{ case.name|upperFirst }}: Bool {
-        get {
             guard case .{{ case.name }} = self else { return false }
```
Comment (BogdanBilonog) [inline PlayerControls/templates/EnumPrism.stencil]: Fix indent

File: PlayerControls/templates/EnumPrism.stencil [hunk 5]
```diff
@@ -46,34 +46,21 @@ import Foundation
...
-        set {
-            guard newValue else { fatalError("Setting false value forbidden") }
-            self = .{{ case.name }}
-        }
     }
 {% endmacro %}
 
 /*For each public enum with cases count > 0 is created extension with cases' prism vars*/
-{% for enum in types.enums where enum.cases.all.count > 0 and not enum.accessLevel == 'private' %}
+{% for enum in types.implementing.GeneratePrism|enum   %}
```
Comment (BogdanBilonog) [inline PlayerControls/templates/EnumPrism.stencil]: Remove extra space after `|enum`

**Pull request merged**